### PR TITLE
feat: user config per service

### PR DESCRIPTION
**Pull request by Kaspiman**
# Reason for This PR

closes: https://github.com/roadrunner-server/roadrunner/issues/1570

## Description of Changes
Added `User` param. [`Group` parameter](https://github.com/roadrunner-server/server/blob/master/plugin.go#L248) does not make sense, although it is present in the current config

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the MIT license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

File: plugin.go [hunk 1]
```diff
@@ -77,12 +77,12 @@ func (p *Plugin) Serve() chan error {
 			procs := value.([]*Process)
 
 			for i := 0; i < len(procs); i++ {
-				cmdStr := procs[i].service.Command
 				err := procs[i].start()
 				if err != nil {
 					errCh <- err
 					return false
 				}
+				cmdStr := procs[i].service.Command
```
Comment (rustatian) [inline plugin.go:85]: The problem of accessing the `proc[i].service.Command` is in the concurrent access to the `service` field. Thus we have to copy it into the `cmdStr` before pushing it to another goroutine. You might see that in the tests.

**Comment by rustatian** [comment]
> [Group parameter](https://github.com/roadrunner-server/server/blob/master/plugin.go#L248) does not make sense, although it is present in the current config

Group actually makes sense and it is used in the `http` plugin to save uploads under the worker's GID/UID (`chown`). By the link you provided it is returned as 0 which means an error/no group.
In the service configuration, specifying the `user` would be sufficient, group would be checked under the hood of the `sdk` `ExecuteFromUser` method.

**Comment by rustatian** [comment]
@Kaspiman Could you please also add docs to this feature?

**Comment by Kaspiman** [comment]
> @Kaspiman Could you please also add docs to this feature?

Yep, need more time in the day. Working on secret improvement for jobs plugin.

**Review by rustatian — APPROVED** [review]
LGTM @Kaspiman 👍🏻

**Pull request merged**